Gradle issue after upgrading to React Native 0.71.7

Hey there, I've been updating our app to the latest stable RN version 0.71.7 and the new Kotlin version doesn't play nicely with this library and throws the following error:
```
FAILURE: Build failed with an exception.

* What went wrong:
The Android Gradle plugin supports only kotlin-android-extensions Gradle plugin version 1.6.20 and higher.
The following dependencies do not satisfy the required version:
project ':amplitude_experiment-react-native-client' -> org.jetbrains.kotlin:kotlin-gradle-plugin:1.5.30

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org/
```
Our solution was to patch the library's `build.gradle` the following way:
```
 buildscript {
-    ext.kotlinVersion = "1.5.30"
     repositories {
         google()
         jcenter()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.3'
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion"
+        classpath 'com.android.tools.build:gradle:7.3.1'
     }
 }
```

This fixed our issue however patching `node_modules` is definitely sub-optimal. The library should be updated to support the latest stable React Native versions. A possible fix could be implemented similarly to how DataDog defensively initializes their kotlin version:
```
 def kotlin_version = rootProject.ext.has('kotlinVersion') ? rootProject.ext.get('kotlinVersion') : project.properties['DdSdkReactNative_kotlinVersion']
...
 implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
```
Thanks!

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -56,5 +56,4 @@ repositories {
 dependencies {
     //noinspection GradleDynamicVersion
     implementation "com.facebook.react:react-native:+"  // From node_modules
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
 }

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -1,5 +1,5 @@
 buildscript {
-    ext.kotlinVersion = rootProject.ext.has('kotlinVersion') ? rootProject.ext.get('kotlinVersion') : "1.5.30"
+    def kotlin_version = rootProject.ext.has('kotlinVersion') ? rootProject.ext.get('kotlinVersion') : "1.5.30"
     repositories {
         google()
         mavenCentral()
@@ -7,7 +7,7 @@ buildscript {
 
     dependencies {
         classpath 'com.android.tools.build:gradle:3.5.3'
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion"
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
     }
 }
 
@@ -56,4 +56,5 @@ repositories {
 dependencies {
     //noinspection GradleDynamicVersion
     implementation "com.facebook.react:react-native:+"  // From node_modules
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
 }

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -1,12 +1,12 @@
 buildscript {
-    def kotlin_version = rootProject.ext.has('kotlinVersion') ? rootProject.ext.get('kotlinVersion') : "1.5.30"
+    def kotlin_version = rootProject.ext.has('kotlinVersion') ? rootProject.ext.get('kotlinVersion') : project.properties['ExperimentReactNativeClient_kotlinVersion']
     repositories {
         google()
         mavenCentral()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.3'
+        classpath 'com.android.tools.build:gradle:7.2.2'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
     }
 }
@@ -19,11 +19,11 @@ def safeExtGet(prop, fallback) {
 }
 
 android {
-    compileSdkVersion safeExtGet('ExperimentReactNativeClient_compileSdkVersion', 29)
-    buildToolsVersion safeExtGet('ExperimentReactNativeClient_buildToolsVersion', '29.0.2')
+    compileSdkVersion safeExtGet('ExperimentReactNativeClient_compileSdkVersion', 31)
+    buildToolsVersion safeExtGet('ExperimentReactNativeClient_buildToolsVersion', '31.0.0')
     defaultConfig {
-        minSdkVersion safeExtGet('ExperimentReactNativeClient_minSdkVersion', 16)
-        targetSdkVersion safeExtGet('ExperimentReactNativeClient_targetSdkVersion', 29)
+        minSdkVersion safeExtGet('ExperimentReactNativeClient_minSdkVersion', 21)
+        targetSdkVersion safeExtGet('ExperimentReactNativeClient_targetSdkVersion', 31)
         versionCode 1
         versionName "1.0"
 

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -1,5 +1,5 @@
 buildscript {
-    ext.kotlinVersion = "1.5.30"
+    ext.kotlinVersion = rootProject.ext.has('kotlinVersion') ? rootProject.ext.get('kotlinVersion') : "1.5.30"
     repositories {
         google()
         mavenCentral()

--- a/android/gradle.properties
+++ b/android/gradle.properties
@@ -1,3 +1,7 @@
+ExperimentReactNativeClient_kotlinVersion=1.6.21
+ExperimentReactNativeClient_compileSdkVersion=31
+ExperimentReactNativeClient_buildToolsVersion=31.0.0
+ExperimentReactNativeClient_targetSdkVersion=31
 android.useAndroidX=true
 android.enableJetifier=true
 kotlin.code.style=official

